rasa/shared/core/events.py

```diff
@@ -314,21 +317,30 @@ def apply_to(self, tracker: "DialogueStateTracker") -> None:
 
     def __eq__(self, other: Any) -> bool:
         """Compares object with other object."""
-        if not isinstance(other, self.__class__):
-            return NotImplemented
-
-        return True
+        # Every class should implement this
+        raise NotImplementedError()
 
     def __str__(self) -> Text:
         """Returns text representation of event."""
         return f"{self.__class__.__name__}()"
 
 
-# noinspection PyProtectedMember
+class EventWithoutExtraAttributes(Event, ABC):
```

Comment:
I find this name is confusing and likely to form a grab bag. Maybe something along the lines of `AlwaysEqualEventMixin`?

Reply:
Mhm, it's not always equal. It's only equal if it has the same type 🤔 In my opinion it'd actually be ok if it ends up as grab bag, e.g. I see that most of this classes have `as_story_string` implemented with `return self.type_name`

Reply:
actually I like your name suggestion, will go with it 👍🏻

Reply:
I do see your point though! If we add more functionality to it we can change the name again to fit.

Reply:
Yep 👍🏻 Let's keep it narrow for now. We can't fix all problems in the `e2e` branch anyway